robolectric/src/main/java/org/robolectric/internal/bytecode/MethodRef.java

```diff
@@ -0,0 +1,44 @@
+package org.robolectric.internal.bytecode;
+
+/**
+ * Reference to a specific method on a class.
+ */
+public class MethodRef {
+  public final String className;
+  public final String methodName;
+
+  public MethodRef(Class<?> clazz, String methodName) {
+    this(clazz.getName(), methodName);
+  }
+
+  public MethodRef(String className, String methodName) {
+    this.className = className;
+    this.methodName = methodName;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+
+    MethodRef methodRef = (MethodRef) o;
+
+    if (!className.equals(methodRef.className)) return false;
+    if (!methodName.equals(methodRef.methodName)) return false;
+
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
+    return className.hashCode();
```

Comment:
Nice catch, I think I just copied this from https://github.com/robolectric/robolectric/pull/1840/files#diff-c8466bf544b1d1b57e346409f9ab80ccL322 so this might be a bug.

That said it wouldn't cause a difference in behavior as hashCode() still works, just gives more collisions. I'll fix it.